aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java

```diff
@@ -149,6 +150,15 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
           fullTableName, persistFailure);
       commitStatus = checkCommitStatus(newMetadataLocation, metadata);
```

Comment:
`checkCommitStatus` is a heavy operation, I think the check of AwsServiceException can happen before that?